documentation/sphinx/source/developer-guide.rst

```diff
@@ -756,6 +756,118 @@ If you only need to detect the *fact* of a change, and your response doesn't dep
+If we merge all overlapping/adjacent intervals then sort, we end up with a canonical representation of this set of keys.
+
+We encode this canonical set as ordered key value pairs like this::
+
+  <namespace><begin1> -> "1"
+  <namespace><end1> -> "0"
+  <namespace><begin2> -> "1"
+  <namespace><end2> -> "0"
+  ...
+
+Python example::
+
+  >>> tr = db.create_transaction()
+  >>> tr.add_read_conflict_key('foo')
+  >>> tr.add_read_conflict_range('bar/', 'bar0')
+  >>> for k, v in tr.get_range_startswith('\xff\xff/transaction/read_conflict_range/'):
+  ...     print(k, v)
+  ...
+  ('\xff\xff/transaction/read_conflict_range/bar/', '1')
+  ('\xff\xff/transaction/read_conflict_range/bar0', '0')
+  ('\xff\xff/transaction/read_conflict_range/foo', '1')
+  ('\xff\xff/transaction/read_conflict_range/foo\x00', '0')
+
+For read-your-writes transactions, this canonical encoding of conflict ranges
+is already available in memory, and so requesting small ranges is
+correspondingly cheaper than large ranges.
+
+For transactions with read-your-writes disabled, this canonical encoding is computed on
+every read, so you're paying the full cost in CPU time whether or not you
+request a small range.
+
+The namespaces for sets of keys are
+
+#. ``\xff\xff/transaction/read_conflict_range/`` This is the set of keys that will be used for read conflict detection. If another transaction writes to any of these keys after this transaction's read version, then this transaction won't commit.
+#. ``\xff\xff/transaction/write_conflict_range/`` This is the set of keys that will be used for write conflict detection. Keys in this range may cause other transactions which read these keys to abort if this transaction commits.
+#. ``\xff\xff/transaction/conflicting_keys/`` If this transaction failed due to a conflict, it must be the case that some transaction attempted [#conflicting_keys]_ to commit with a write conflict range that intersects this transaction's read conflict range. This is the subset of your read conflict range that actually intersected a write conflict from another transaction.
+
+Caveats
+~~~~~~~
+
+#. ``\xff\xff/transaction/read_conflict_range/`` The conflict range for a read is sometimes not known until that read completes (e.g. range reads with limits, key selectors). When you read from these special keys, the returned future first blocks until all pending reads are complete so it can give an accurate response.
+#. ``\xff\xff/transaction/write_conflict_range/`` The conflict range range for a ``set_versionstamped_key`` atomic op is not known until commit time. You'll get an approximate range (the actual range will be a subset of the approximate range) until the precise range is known.
+#. ``\xff\xff/transaction/conflicting_keys/`` Since using this feature costs server resources, it's disabled by default. You must opt in by setting the ``report_conflicting_keys`` transaction option.
```

Comment:
`costs server resources` --> `costs server (i.e., proxy and resolver) resources`, so that users have more info about how to configure their cluster if this key reading becomes a problem.